src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -93,6 +95,9 @@
     private static final String PREBID_EXT = "prebid";
     private static final String CACHE = "cache";
     private static final String DEFAULT_CURRENCY = "USD";
+    private static final String EVENT_CALLBACK_URL_PATTERN = "%s/event?type=%s&bidid=%s&bidder=%s";
```

Comment:
What about extracting event construction logic (even though it's tiny) to a separate service?